docs/dev/reader-concurrency-semaphore.md

```diff
@@ -59,6 +59,7 @@ Permits are in one of the following states:
 * `active/await` - a previously `active/need_cpu` permit, which needs something other than CPU to proceed, it is waiting on I/O or a remote shards, other permits can be admitted while the permit is in this state, pending resource availability;
 * `inactive` - the permit was marked inactive, it can be evicted to make room for admitting more permits if needed;
 * `evicted` - a former inactive permit which was evicted, the permit has to undergo admission again for the read to resume;
+* `aborted` - the permit was rejected from execution during admission as it was detected the read might time out during execution;
```

Comment:
We should make the language more general here, so this state covers all cases where a read is aborted: e.g. timing out in the middle.

Reply:
The description is correct for the given implementation. I do not change permit's state when it times out. I left it as it originally was, but of course that can be changed to make it more verbose.

Reply:
Right. I think `on_timeout()` should call `on_abort()`, but I don't insist you do it in this PR, it can be done later as a cleanup.